src/actions/textColor.ts

```diff
@@ -2,46 +2,41 @@ import _ from 'lodash'
 import State from '../@types/State'
 import Thunk from '../@types/Thunk'
 import setDescendant from '../actions/setDescendant'
-import themeColors from '../selectors/themeColors'
+import * as selection from '../device/selection'
+import pathToThought from '../selectors/pathToThought'
 import reducerFlow from '../util/reducerFlow'
 import deleteAttribute from './deleteAttribute'
 
 /** Sets the text color or background color of the cursor. */
 const textColor = (
   state: State,
-  { backgroundColor, color, shape }: { backgroundColor?: string; color?: string; shape?: 'bullet' | 'text' },
+  {
+    backgroundColor,
+    color,
+    shape,
+    fullySelected,
+  }: { backgroundColor?: string; color?: string; shape?: 'bullet' | 'text'; fullySelected?: boolean },
 ) => {
   if (!state.cursor) return state
   const path = state.cursor
-  const colors = themeColors(state)
-
-  return reducerFlow([
-    // set bullet to text color
-    color && color !== 'default'
-      ? setDescendant({ path, values: ['=bullet', '=style', 'color', backgroundColor || color] })
-      : deleteAttribute({ path, values: ['=bullet', '=style', 'color'] }),
-
-    // set text color
-    // clear color if white
-    backgroundColor || color !== 'default'
-      ? setDescendant({ path, values: ['=style', 'color', color || 'rgba(0, 0, 0, 1)'] })
-      : deleteAttribute({ path, values: ['=style', 'color'] }),
-
-    // set background color
-    // clear background color if default or unset
-    backgroundColor && backgroundColor !== colors.bg
-      ? setDescendant({
-          path,
-          values: ['=styleAnnotation', 'backgroundColor', backgroundColor === 'inverse' ? colors.fg : backgroundColor],
-        })
-      : deleteAttribute({ path, values: ['=styleAnnotation', 'backgroundColor'] }),
-  ])(state)
+  // set bullet to text color when the entire thought selected
+  const newState =
+    fullySelected && ((color && color !== 'default') || (backgroundColor && backgroundColor !== 'inverse'))
+      ? [setDescendant({ path, values: ['=bullet', '=style', 'color', backgroundColor! || color!] })]
+      : [deleteAttribute({ path, values: ['=bullet', '=style', 'color'] })]
+  return reducerFlow(newState)(state)
```

Comment:
There's no need to use `reducerFlow` any more, as we are no longer composing multiple reducers. You can call the reducers directly:

```ts
return fullySelected && ((color && color !== 'default') || (backgroundColor && backgroundColor !== 'inverse'))
      ? setDescendant(state, { path, values: ['=bullet', '=style', 'color', backgroundColor! || color!] })
      : deleteAttribute(state, { path, values: ['=bullet', '=style', 'color'] })
```